ush/ioda/bufr2ioda/run_bufr2ioda.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+import argparse
+import os
+from gen_bufr2ioda_json import gen_bufr_json
+from wxflow import Logger, Executable, cast_as_dtype
+
+# Initialize root logger
+logger = Logger('run_bufr2ioda.py', level='INFO', colored_log=True)
+
+
+def bufr2ioda(CDATE, RUN, DMPDIR, config_template_dir, COM_OBS):
+    logger.info(f"Process {CDATE} {RUN} from {DMPDIR} to {COM_OBS} using {config_template_dir}")
+
+    # Derived parameters
+    PDY = CDATE[:8]
+    cyc = CDATE[8:10]
+
+    # Load configuration
+    config = {
+        'RUN': RUN,
+        'PDY': PDY,
+        'cyc': cyc,
+        'DMPDIR': DMPDIR,
+        'COM_OBS': COM_OBS
+    }
+    config['current_cycle'] = cast_as_dtype(CDATE)
+
+    # Get gdasapp root directory
+    DIR_ROOT = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../.."))
+    USH_IODA = os.path.join(DIR_ROOT, "ush", "ioda", "bufr2ioda")
+
+    # Create output directory if it doesn't exist
+    os.makedirs(COM_OBS, exist_ok=True)
+
+    # Specify observation types to be processed by a script
+    BUFR_py = ["satwind_amv_goes"]
+
+    for obtype in BUFR_py:
+        logger.info(f"Convert {obtype}...")
+        json_output_file = os.path.join(COM_OBS, f"{obtype}_{PDY}{cyc}.json")
+        template = config_template_dir + '/bufr2ioda_' + obtype + '.json'
+        gen_bufr_json(config, template, json_output_file)
+
+        # Use the converter script for the ob type
+        bufr2iodapy = USH_IODA + '/bufr2ioda_' + obtype + ".py"
+        cmd = Executable(bufr2iodapy)
+        cmd.add_default_arg('-c')
+        cmd.add_default_arg(json_output_file)
+        logger.info(f"Executing {cmd}")
+        cmd()
+
+        # Check if the converter was successful
+        if os.path.exists(json_output_file):
+            os.remove(json_output_file)
+        else:
+            print(f"Problem running converter script for {obtype}")
```

Comment:
Suggestion accepted at cff2ce5.

Reply:
Instead of this `if-else`, you should use `wxflow.rm_p`.
It not only traps any OSerror it encounters, but will also check if the file is present.
I am just alerting of the tools we have to handle some of these basic functions without having to write too much individual code.

Reply:
Suggestion accepted at c0c0910.